eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -54,9 +61,20 @@ class ReindexCommand extends ContainerAwareCommand
     public function initialize(InputInterface $input, OutputInterface $output)
     {
         parent::initialize($input, $output);
-        $this->searchIndexer = $this->getContainer()->get('ezpublish.spi.search.indexer');
+
+        $searchIndexer = $this->getContainer()->get('ezpublish.spi.search.indexer');
+        // It's required to clone $searchIndexer, otherwise global service state will be changed due to setting errorCollector
+        $this->searchIndexer = clone $searchIndexer;
```

Comment:
For me this is kind of design smell, but unsure how much more it's needed to iterate on this.

One way to avoid would be if continue-on-error knowledge is moved to indexer, so all the stuff on bool return value on ErrorCollector interface and such can be removed simplifying it to truly be only about collecting errors.

And instead end up with `->updateSearchIndex($contentIds, $commit, $continueOnError)` so we don't need to clone anything as it will more cleanly be a normal call argument like it should be.

But it's a new argument, so A. either needs to be introduced as commented out `/*bool $continueOnError= false*/` and with php doc stating it will be added to interface in next major _(with inline deprecation notice or something we can find when next major comes)_. Symfony sometimes does this on addition/removal of arguments, so for me this is fine, impl can always add arguments interface does not have as long as it has default value.

Or B. we need to add yet another interface and deprecate the old one.

I think I prefer A as otherwise we keep on adding new interfaces all the time, and it is a bit softer dependency for dealing with kernel vs solr bundle.